common/src/main/java/com/wynntils/features/trademarket/TradeMarketQuickSearchFeature.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.trademarket;
+
+import com.mojang.blaze3d.platform.InputConstants;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.consumers.features.Feature;
+import com.wynntils.core.consumers.features.properties.RegisterKeyBind;
+import com.wynntils.core.keybinds.KeyBind;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.config.Category;
+import com.wynntils.core.persisted.config.Config;
+import com.wynntils.core.persisted.config.ConfigCategory;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.mc.event.MenuEvent.MenuClosedEvent;
+import com.wynntils.mc.event.ScreenClosedEvent;
+import com.wynntils.mc.event.ScreenOpenedEvent;
+import com.wynntils.models.containers.containers.TradeMarketContainer;
+import com.wynntils.utils.mc.KeyboardUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.ChatScreen;
+import net.minecraft.world.inventory.Slot;
+import net.neoforged.bus.api.EventPriority;
+import net.neoforged.bus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.TRADEMARKET)
+public class TradeMarketQuickSearchFeature extends Feature {
+    @Persisted
+    public final Config<Boolean> instantSearch = new Config<>(true);
+
+    @Persisted
+    public final Config<Boolean> hidePrompt = new Config<>(true);
+
+    @Persisted
+    public final Config<Boolean> autoCancel = new Config<>(true);
+
+    @RegisterKeyBind
+    private final KeyBind quickSearchKeyBind = new KeyBind(
+            "Quick Search TM",
+            GLFW.GLFW_MOUSE_BUTTON_MIDDLE,
+            InputConstants.Type.MOUSE,
+            true,
+            null,
+            this::tryQuickSearch);
+
+    private static final Pattern TYPE_TO_CHAT_PATTERN = Pattern.compile(
+            "^§5(\uE00A\uE002|\uE001) \n\uE001 Type the .* or type (\n\uE001 'cancel' to|'cancel' to \n\uE001) cancel:\n\uE001 ");
```

Comment:
The pattern from `TradeMarketAutoOpenChatFeature` is a little nicer so I'd say use that here too. Then use `StyledTextUtils.unwrap(event.getOriginalStyledText()).stripAlignment()` to check for the match.

Annoying to duplicate the pattern but as mentioned before ideally this should all be handled in the model

Reply:
[6653343](https://github.com/Wynntils/Wynntils/pull/2908/commits/6653343f32347c662583cc701210cbf9074727e2)